XenAdmin/Wizards/PatchingWizard/PlanActions/UploadPatchToMasterPlanAction.cs

```diff
@@ -90,14 +92,9 @@ protected override void RunWithSession(ref Session session)
                         throw new Exception(Messages.ACTION_UPLOADPATCHTOMASTERPLANACTION_FAILED);
                     }
 
-                    var newMapping = new PoolPatchMapping()
-                    {
-                        MasterHost = Helpers.GetMaster(session.Connection),
-                        XenServerPatch = patch,
-                        Pool_patch = poolPatch
-                    };
+                    var newMapping = new PoolPatchMapping(patch, poolPatch, Helpers.GetMaster(session.Connection));
 
-                    if (!mappings.Any(m => m.MasterHost == newMapping.MasterHost && m.Pool_patch == newMapping.Pool_patch && m.XenServerPatch == patch))
+                    if (!mappings.Contains(newMapping))

```

Comment:
Contains will use the Equals from the PoolPatchMapping class which is not comparing uuids, so you'll have the same problem